bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -249,22 +258,15 @@ private synchronized void updateDailyWeeklyData() {
      */
     private synchronized void updateMonthlyData() {
         if (isLinked(LAST_MONTH) || isLinked(THIS_MONTH)) {
-            cachedMonthlyData.getValue().ifPresentOrElse(values -> {
-                Aggregate months = values.aggregats.months;
-                updateKwhChannel(LAST_MONTH, months.datas.get(0));
-                if (months.datas.size() > 1) {
-                    updateKwhChannel(THIS_MONTH, months.datas.get(1));
-                } else {
-                    updateKwhChannel(THIS_MONTH, 0.0);
-                }
-            }, () -> {
-                if (ZonedDateTime.now().getDayOfMonth() == 1) {
-                    updateKwhChannel(THIS_MONTH, 0.0);
-                    updateKwhChannel(LAST_MONTH, Double.NaN);
-                } else {
-                    updateKwhChannel(THIS_MONTH, Double.NaN);
-                }
-            });
+            cachedMonthlyData.getValue().ifPresentOrElse(
+                    values -> setCurrentAndPrevious(values.aggregats.months, THIS_MONTH, LAST_MONTH), () -> {
```

Comment:
If I correctly understand, the storing of values has been reversed, the value for the current month was previously the value at index 0, it is now the value at biggest index ?

Reply:
You're correct, data are now provided in ascending order.